lib/modules/datasource/deb/packages.ts

```diff
@@ -0,0 +1,227 @@
+import upath from 'upath';
+import { logger } from '../../../logger';
+import * as fs from '../../../util/fs';
+import { toSha256 } from '../../../util/hash';
+import type { Http } from '../../../util/http';
+import { escapeRegExp, regEx } from '../../../util/regex';
+import { joinUrlParts } from '../../../util/url';
+import { computeFileChecksum } from './checksum';
+import { cacheSubDir } from './common';
+import { extract, getFileCreationTime } from './file';
+import { getReleaseFileContent } from './release';
+import { PackagesCompressionAlgos } from './types';
+import { getBaseSuiteUrl, getPackagePath } from './url';
+
+/**
+ * Downloads and extracts a package file from a component URL.
+ *
+ * @param componentUrl - The URL of the component.
+ * @param http - The HTTP client to use for downloading.
+ * @returns The path to the extracted file and the last modification timestamp.
+ * @throws Will throw an error if no valid compression method is found.
+ */
+export async function downloadAndExtractPackage(
+  componentUrl: string,
+  http: Http,
+): Promise<{ extractedFile: string; lastTimestamp: Date }> {
```

Comment:
please create an interface for such things, especially if the function is exported.